### DISCUSSION TITLE
Make `adjust-cursor-thickness` apply to `cursor-style = underline` too

### DISCUSSION THREAD

**Discussion by zx8** [ghostty-org/ghostty#3194] [Feature Requests, Ideas] [answered]
Coming from [this answer](https://github.com/ghostty-org/ghostty/discussions/2657#discussioncomment-11257924), it appears this option already applies to the thickness of the cursor when the style is set to `bar`, but it does not apply when the cursor style is set to `underline`.

Here's how it looks between iTerm (left) and Ghostty (right) at the moment:

<img width="200" alt="image" src="https://github.com/user-attachments/assets/b7311c42-fb58-4dca-8475-1c8430f578f0" />

Would it be possible to extend this setting to work with `underline` mode too?

**Comment by gko** [comment]
to adjust underline cursor thickness you can use `adjust-underline-thickness`:
https://ghostty.org/docs/config/reference#adjust-underline-thickness

although, it would also impact other underlined text

**Reply by zx8** [reply]
Yeah, I know. I don't really want to affect general underline thickness; this request is specifically related to cursor thickness.

**Comment by xzbdmw** [comment]
This is important in vim operator pending mode, it's surprised there is not a dedicated option for it.

**Comment by doums** [comment]
`adjust-underline-thickness` is great to adjust general underlined text style. On the other hand the cursor thickness is handled by a dedicated option `adjust-cursor-thickness` which is good. Having a separate option to adjust the cursor thickness regardless of whatever other thickness options is nice. (And what I was looking for).
But the problem is: this option only applies to the cursor vertical 'bar' style while for some reason, the cursor horizontal bar style (aka 'underline') is still controlled by `adjust-underline-thickness`.
In my case I need a cursor horizontal bar thicker than the general underlined text style. But yea it is impossible currently.
Also surprised by the options design here.

**Comment by qwerasd205** [comment]
Yeah, currently we don't have a dedicated function for drawing the underline cursor, we use the same code as draws the actual underline. Separating it out should be easy enough.

**Reply by qwerasd205** [reply, marked as answer]
Created issue #7651 for this.
